app/src/main/java/ch/epfl/sdp/blindly/main_screen/match/MatchPageFragment.kt

```diff
@@ -300,4 +324,27 @@ class MatchPageFragment : Fragment(), CardStackListener {
 
         return thisLoc.distanceTo(otherLoc).roundToInt() / M_TO_KM
     }
+
+    /**
+     * When a user likes someone, check if the other liked them
+     * too and match them both
+     *
+     */
+    @RequiresApi(Build.VERSION_CODES.N)
+    private suspend fun checkMatch() {
+        val otherUser = userRepository.getUser(likedUserId)
+        if (otherUser?.likes?.contains(currentUserId)!!) {
+            userRepository.updateProfile(
+                likedUserId,
+                "matches",
+                otherUser.matches?.plus(currentUserId)
+            )
+            userRepository.updateProfile(
+                currentUserId,
+                "matches",
+                currentUser.matches?.plus(likedUserId)
+            )
+            Toast.makeText(context, "It's a match !", Toast.LENGTH_SHORT).show()
```

Comment:
Even the notification is there :D Nice!!

Reply:
Yeah, but it's more a temporary notification xD it's only sent to the user 'provoking' the like so I'm not sure we should keep it haha

Reply:
Yeah true, the my matches page actually have a listener of matches of Firestore, maybe we can move this there.